Домашнее задание 11

v-nvtsk opened the pull request:
создать репозиторий на гитхабе
инициировать проект в репозитории
решить предложенные задачи
сделать коммит (а лучше несколько - по одному на задание)
открыть пуллреквест
прислать ссылку на пуллревест в чат по дз

v-nvtsk committed:
feat: add packages and config files

v-nvtsk committed:
feat: add config files to ESLint ignore

v-nvtsk committed:
feat: rename JEST config file

v-nvtsk committed:
feat: add Jest plugin to eslint config

v-nvtsk committed:
fix: change eslintignore file with eslint config

Remove .eslintignore file and move ignored list
to .eslint config file section "ignorePattern"

v-nvtsk committed:
fix: update jest configuration:

rename jest.config.mjs to .js
remove jsdom package
install babel packages
add babel.config.js file

v-nvtsk committed:
feat: add test:watch script

v-nvtsk committed:
feat: add lint-staged script

v-nvtsk committed:
feat: install package eslint-plugin-jest

v-nvtsk committed:
feat: change lint-staged config in package.json

v-nvtsk committed:
update package-lock.json

v-nvtsk committed:
Fix husky pre-commit to run lint-staged

v-nvtsk committed:
Fix husky pre-commit: move npm test to lint-staged

v-nvtsk committed:
dz1-6 tests: init tests
add functions import
add check for function to be function

v-nvtsk committed:
dz1-6 tests: add mock for console.log

v-nvtsk committed:
dz1-6 tests: add tests for dz1-1

v-nvtsk committed:
dz1-6 tests: add tests for dz1-2

v-nvtsk committed:
dz1-6 tests: add tests for dz1-3

v-nvtsk committed:
dz1-6 tests: add tests for dz2

v-nvtsk committed:
dz1-6 tests: add tests for dz3

v-nvtsk committed:
dz1-6 tests: add tests for dz4

v-nvtsk committed:
dz1-6 tests: add tests for dz5

v-nvtsk committed:
dz1-6 tests: add tests for dz6

v-nvtsk committed:
dz7 tests: add tests for dz7

v-nvtsk committed:
dz8 tests: add tests for dz8

v-nvtsk committed:
dz9 tests: add tests for dz9

v-nvtsk committed:
dz10 tests: add tests for dz10

v-nvtsk committed:
add index.html

v-nvtsk committed:
add main.js

v-nvtsk committed:
add dz 1-6

v-nvtsk committed:
add dz 7

v-nvtsk committed:
add dz 8

v-nvtsk committed:
add dz 9

v-nvtsk committed:
add dz 10

v-nvtsk committed:
Update deploy-gh-pages.yaml

v-nvtsk committed:
Add file extensions to import in main.js

v-nvtsk committed:
Merge branch 'js' of github.com:v-nvtsk/otus-jsbasic-dz11 into js

v-nvtsk committed:
fix lint-staged commands in config

v-nvtsk committed:
fix dz1-6 functions export from commonjs to ESM

v-nvtsk committed:
fix dz7 functions export from commonjs to ESM

v-nvtsk committed:
fix dz8 functions export from commonjs to ESM

v-nvtsk committed:
fix dz9 functions export from commonjs to ESM

v-nvtsk committed:
fix dz10 functions export from commonjs to ESM

v-nvtsk committed:
refactor main.js: add createContainer

add createContainer function to create elements without
elements in index.html
change function calls for containers
remove template container elements from index.html

v-nvtsk committed:
add missed dz10 call

v-nvtsk committed:
Add header creation for dz10

v-nvtsk committed:
add tests for createContainer in main.js

v-nvtsk committed:
refactor: dz1_2 add default parameter values

and remove excessive variables

v-nvtsk committed:
refactor: dz1_1 add implicit type coercion

v-nvtsk committed:
fix testData for dz2_1 and dz6_diff

vvscode commented on package.json:
```diff
@@ -0,0 +1,60 @@
+    "jest-environment-jsdom": "^29.7.0",
+    "lint-staged": "^15.0.2",
+    "prettier": "^3.0.3",
+    "stylelint": "^15.11.0",
```
зачем он в этом задании?

v-nvtsk replied:
В этом задании не нужен

vvscode commented on package.json:
```diff
@@ -0,0 +1,60 @@
+  "author": "Novitskiy Vladislav",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/v-nvtsk/otus-jsbasic-git/issues"
```
для чего тут (и выше) неправильная ссылка?

v-nvtsk replied:
Ссылка поменялась в процессе работы. Исправил

vvscode commented on package.json:
```diff
@@ -0,0 +1,60 @@
+    "lint-staged": "lint-staged",
+    "lint": "eslint ./src",
+    "lint:fix": "eslint ./src --fix",
+    "lint:prod": "eslint ./src --fix --color --rule \"{\"no-console\":\"error\"}\"",
```
а давай правила настраивать через файл конфига, а не через скрипты

vvscode replied:
дальше - для чего тут `--fix` ? я вообще не вижу, где скрипт используется, но если он для CI - то тут задача должна падать, а не исправлять ошибки

v-nvtsk replied:
Идея в том, чтобы линтер меня не мучал сообщениями о console.log в процессе разработки, но запрещал на этапе коммита

v-nvtsk replied:
> дальше - для чего тут `--fix` ? я вообще не вижу, где скрипт используется, но если он для CI - то тут задача должна падать, а не исправлять ошибки

Изначально запускал его в хуке pre-commit, но в данном задании он не позволял коммитить задачи с console.log
Использую сейчас только для запуска вручную.

vvscode replied:
а как он тебя мучает на этапе разработки?

v-nvtsk replied:
куча предупреждений о наличии console.log в коде
в данном случае можно конечно отдельным правилом отключить в начале файла
а в общем случае не хочется пушить с отладочными console.log.

vvscode commented on package.json:
```diff
@@ -0,0 +1,60 @@
+  "scripts": {
+    "start": "node ./src/index.js",
+    "test": "jest",
+    "test:watch": "clear && node --experimental-vm-modules --no-warnings node_modules/jest/bin/jest.js --watch --verbose",
```
для чего нужно включать `experimental-vm-modules` если у тебя уже есть babel? тем более, что для обычного теста ты их не включаешь, и по итогу у тебя гоняются разные сценарии

v-nvtsk replied:
Включал до того, как установил babel.
Исправил локально, но не запушил

vvscode commented on .eslintrc:
```diff
@@ -0,0 +1,53 @@
+    "node_modules",
+    ".vscode",
+    ".git",
+    "src/__tests__",
```
такой папки нет

но даже, если бы она была - какая причина не проверять тесты?

v-nvtsk replied:
упоминание несуществующей папки удалил.
где-то на просторах интернета попадалось, что тесты добавлять в игнор линтера.

vvscode commented on .eslintrc:
```diff
@@ -0,0 +1,53 @@
+    ".vscode",
+    ".git",
+    "src/__tests__",
+    "jest.config.js"
```
и какая причина не проверять конфиг?

v-nvtsk replied:
очень длинные комменты, на которые линтер ругается, а исправлять (удалять) их не хочется

vvscode replied:
ну так это не повод отключать, а повод сказать правилу, что длинна строки в комментах не учитывается

https://eslint.org/docs/latest/rules/max-len#ignorecomments

v-nvtsk replied:
устранил

vvscode commented on .prettierrc:
```diff
@@ -0,0 +1,12 @@
+{
+  "printWidth": 120,
```
вообще идея преттира как раз в том, чтобы его не настраивать

v-nvtsk replied:
Не понял мысль.
Конкретно с этой настройкой хотел удлинить строки, такая же настройка в линтере.
Если не добавить эту настройку претир после линтера будет опять обрезать строки до 80 (емнип) символов.

vvscode commented on src/dz10.js:
```diff
@@ -0,0 +1,30 @@
+  })
+
+}
+export default validateString;
```
судя по завершениям файлов - преттир их не обрабатывал, от чего?

v-nvtsk replied:
Не указал плагин в настройках линтера. Исправил локально, но не пушил. Не уверен как правильно это делать, ветка другая

vvscode commented on src/dz1_6.test.js:
```diff
@@ -0,0 +1,375 @@
+  dz1_1, dz1_2, dz1_3, dz2_1, dz2_2, dz2_3,
+  dz3_1, dz3_2, dz3_3,
+  dz4, dz5, dz6_diff, dz6_isWord, dz6_pow
+} from './dz1_6';
```
это очень странная запись, если честно

v-nvtsk replied:
Ну я сразу писал, что в первые шесть заданий в одном файле.
Да и тематически они почти все на console.log завязаны.
Если надо исправить - скажите, пожалуйста, как?
Разбивать на отдельные файлы?

vvscode replied:
да нет, я про саму запись

ну и если что - можно было
 

`import * as dz from './dz1_6'`

vvscode commented on src/dz1_6.test.js:
```diff
@@ -0,0 +1,375 @@
+  dz4, dz5, dz6_diff, dz6_isWord, dz6_pow
+} from './dz1_6';
+
+describe("dz 1 to 6 test suite", () => {
```
и тк этот describe ничего не добававляет - его можно было бы просто убрать

v-nvtsk replied:
в него поместил beforeEach afterEach. Вне describe они вроде бы не работают.

vvscode replied:
работают

v-nvtsk replied:
устранил

vvscode commented on package.json:
```diff
@@ -0,0 +1,60 @@
+    "stylelint-config-standard-scss": "^11.1.0"
+  },
+  "lint-staged": {
+    "*.{cjs,mjs,ts,tsx,js,jsx}": [
```
`yaml` ? `json` ? `*rc` ?

v-nvtsk replied:
Ну, изначально хотел сделать универсальную сборку. На вырост)

v-nvtsk committed:
fix: urls in package.json

v-nvtsk committed:
remove package stylelint from project

v-nvtsk committed:
fix: add prettier plugin in .eslintrc

v-nvtsk committed:
fix .eslintrc: remove tests from ignorelist

v-nvtsk committed:
remove unused file extensions for lint-staged

leave only .js for lint-staged config

vvscode commented on src/dz9.test.js:
```diff
@@ -0,0 +1,118 @@
+    { a: 1, b: 2, c: 3, result: 'No roots' } // x^2 + 2x + 3 = 0 (Complex roots)
+  ];
+
+  const parent = document.createElement('div');
+  document.body.append(parent);
+
+  quadricEquation(parent);
+
+  const inputA = parent.querySelector('input.inputA');
+  const inputB = parent.querySelector('input.inputB');
+  const inputC = parent.querySelector('input.inputC');
+
+  const button = parent.querySelector('button');
+  const resultString = parent.querySelector('p')
```
воот это все хорошо бы в before* вынести

vvscode commented on src/dz10.js:
```diff
@@ -0,0 +1,30 @@
+  button.addEventListener('click', () => {
+    const str = input.value;
+
+    if (/\d{1,2}\.\d{1,2}\.\d{4}/gi.test(str)) {
```
а по каким-то другими фильтрами (более надежными и читаемыми) нельзя обойтись?

vvscode commented on .husky/pre-push:
```diff
@@ -0,0 +1,32 @@
+echo -e "===\n>> Pre-push Hook:: Checking branch name ..."
+branchName=$(git rev-parse --abbrev-ref HEAD)
+
+if [ $branchName == 'master']; then
```
https://www.npmjs.com/package/branch-name-lint

v-nvtsk committed:
fix: remove jest.config from lint ignore

add es-lint rule to disable max-len in jest.config.js

v-nvtsk committed:
fix lint-staged config

remove incorrect braces around single file extension

v-nvtsk committed:
fix dz1_6.test.js remove common describe()

v-nvtsk committed:
fix: remove jest.config from eslint ignore

v-nvtsk committed:
dz7 test remove document appending

v-nvtsk committed:
dz 8 test refactor

remove document appending
move test setup to beforeEach

v-nvtsk committed:
dz8.test.js restyle using pretter manually

v-nvtsk committed:
dz7.test.js restyle using pretter manually

v-nvtsk committed:
dz 9 test refactor

remove document appending
move tests setup to beforeEach

v-nvtsk committed:
dz9.test.js restyle using prettier manually

v-nvtsk committed:
dz10 refactor

rename str variable to inputString
create validator functions and move logic there
isDate is now really validate date

v-nvtsk committed:
dz10 test add testData for date variations

v-nvtsk committed:
dz10 test move test setup to beforeEach

v-nvtsk committed:
dz1_6.js restyle using prettier manually

v-nvtsk committed:
js files restyle using prettier manually

 Changes to be committed:
	modified:   src/dz10.js
	modified:   src/dz10.test.js
	modified:   src/dz1_6.test.js
	modified:   src/dz7.js
	modified:   src/dz8.js
	modified:   src/dz9.js
	modified:   src/main.js
	modified:   src/main.test.js

v-nvtsk committed:
codesandbox-link-comment action update

add lines with comment to view gh-pages branch
 on codesandbox directly with the link provided
